tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td

```diff
@@ -760,17 +760,28 @@ foreach ArgMinMaxOp = [TFL_ArgMinOp, TFL_ArgMaxOp] in {
      (AxesIsLastDimension $axes, $logits)]>;
 }
 
-class AllElementsAreI64<string val> : Constraint<CPred<
-  "($0.isa<DenseElementsAttr>() && "
-   "$0.cast<DenseElementsAttr>().getType().cast<ShapedType>().getElementType().isInteger(64) && "
-   "std::all_of($0.cast<DenseElementsAttr>().getValues<int64_t>().begin(), "
-               "$0.cast<DenseElementsAttr>().getValues<int64_t>().end(), "
-               "[](int64_t v){ return v == " #val# ";}))">>;
-
-// Remove Slice ops slicing the whole input tensor, effectively no-op
+class AllElementsAreI32OrI64<string val> : Constraint<CPred<
```

Comment:
AllElementsAreI32OrI64 and IsSliceWholeInput are complicated. It would be better to move the logic under the optimize.cc for the readability.